src/dlgConnectionProfiles.cpp

```diff
@@ -1868,8 +1866,6 @@ bool dlgConnectionProfiles::validateProfile()
             notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(), tr("This profile name is already in use.")));
             validName = false;
             valid = false;
-        } else {
-            pItem->setText(name);
```

Comment:
This is the root cause of the problem and the real fix this PR: the item's name was syncronised, as a side-effect of the validation (bad!) with the lineedit's name. Because both were the same, the 'is this a new profile?' detection was broken - and the folder on disk for the new profile was never created.